domain/batch.js

```diff
@@ -21,16 +22,25 @@ class Batch {
         this.filePrefix = filePrefix;
     }
 
-    addDataFile(name, size){
-        this.#addFile(name, size, null, true)
+    addDataFile(name, size, url, dataCommons, studyID){
+        const fileID = this.#generateDataFileUUID(name, url, dataCommons, studyID);
+        this.#addFile(name, size, null, true, fileID);
     }
 
     addMetadataFile(name, size, signedURL){
         this.#addFile(name, size, signedURL, false)
     }
 
-    #addFile(name, size, signedURL, isDataFile) {
-        const file = new BatchFile(name, size, signedURL, this.filePrefix, isDataFile);
+    #generateDataFileUUID(fileName, url, dataCommons, studyID) {
+        const urlUUID = v5(url, v5.URL, undefined, undefined);
+        const dataCommonsUUID = v5(dataCommons, urlUUID, undefined, undefined);
+        const studyUUID = v5(studyID, dataCommonsUUID, undefined, undefined);
+        return `${DG_4DFC}/${v5(fileName, studyUUID, undefined, undefined)}`;
```

Comment:
The prefix "dg.4DFC/" only need to be added if "omit-dcf-prefix" is null or set to false in the datamodel (content.json)